src/ray/raylet/node_manager.cc

```diff
@@ -682,13 +682,21 @@ void NodeManager::HandleActorStateTransition(const ActorID &actor_id,
       SubmitTask(method, Lineage());
     }
   } else if (actor_registration.GetState() == ActorTableData::DEAD) {
-    // When an actor dies, loop over all of the queued tasks for that actor
-    // and treat them as failed.
+    // When an actor dies, loop over (i) all of the queued tasks for that actor
+    // and treat them as failed (ii) all actors and kill those whose parent actor
+    // is the actor that just died.
     auto tasks_to_remove = local_queues_.GetTaskIdsForActor(actor_id);
     auto removed_tasks = local_queues_.RemoveTasks(tasks_to_remove);
     for (auto const &task : removed_tasks) {
       TreatTaskAsFailed(task, ErrorType::ACTOR_DIED);
     }
+    for (auto &actor_entry_it : actor_registry_) {
+      if (actor_entry_it.second.GetParentActorID() == actor_id) {
```

Comment:
I believe you need to check to make sure that this actor is actually local, which you can do by checking the node manager ID. Actually, the current test that you have would probably fail once you simulate it on multiple nodes due to this, so that would be a good regression test.